alfresco-actuators/src/main/java/eu/xenit/actuators/services/AlfrescoInfoService.java

```diff
@@ -0,0 +1,198 @@
+            }
+        }
+
+        return map;
+    }
+
+
+    private LicenseInfo retrieveLicenseInfo() {
+        if(licenseService==null) {
+            licenseService=appContext.getBeansOfType(LicenseService.class).get("licenseService");
+        }
+        LicenseDescriptor licenseDescriptor = descriptorService.getLicenseDescriptor();
+
+        // community
+        if(licenseService == null || licenseDescriptor == null) {
+           return null;
+        }
+
+        return new LicenseInfo()
+                .valid(licenseService.isLicenseValid())
+                .remainingDays(licenseDescriptor.getRemainingDays())
+                .holder(licenseDescriptor.getHolder().getName())
+                .organisation(licenseDescriptor.getHolderOrganisation())
+                .maxUsers(licenseDescriptor.getMaxUsers())
+                .clusteringEnabled(licenseDescriptor.isClusterEnabled())
+                .encryptionEnabled(licenseDescriptor.isCryptodocEnabled())
+                .heartbeatDisabled(licenseDescriptor.isHeartBeatDisabled());
+    }
+
+    // needs authentication
+    private StatusInfo retrieveStatusInfo() {
+        final StatusInfo statusInfo = new StatusInfo();
+
+        retryingTransactionHelper.doInTransaction(new RetryingTransactionHelper.RetryingTransactionCallback<Object>() {
+
+            @Override
+            public Object execute() throws Throwable {
+                AuthenticationUtil.runAs(new AuthenticationUtil.RunAsWork<Object>() {
+
+                    @Override
+                    public Object doWork() throws Exception {
+                        statusInfo
+                                .readOnly(repoAdminService.getUsage().isReadOnly())
+                                .auditEnabled(auditService.isAuditEnabled())
+                                .thumbnailGeneration(thumbnailService.getThumbnailsEnabled());
+
+                        return null;
+                    }
+                }, AuthenticationUtil.getAdminUserName());
+                return null;
+            }
+        },true);
+
+        return statusInfo;
+
+    }
+
+    private List<ModuleInfo> retrieveAlfrescoModules() {
+
+        List<ModuleDetails> allModules = moduleService.getAllModules();
+        List<ModuleInfo> list = new ArrayList<>(allModules.size());
+
+        for(ModuleDetails details : allModules) {
+            list.add(toAlfrescoModule(details));
+        }
+
+        return list;
+    }
+
+    private static ModuleInfo toAlfrescoModule(ModuleDetails moduleDetails) {
+        return new ModuleInfo()
+                .title(moduleDetails.getTitle())
+                .description(moduleDetails.getDescription())
+                .installState(moduleDetails.getInstallState().toString())
+                .version(moduleDetails.getProperties().getProperty(ModuleDetails.PROP_VERSION));
+    }
+
+    @Override
+    public boolean isHealthy() {
+        getAlfrescoInfo();
+        return true;
```

Comment:
so why do we actually call `getAlfrscoInfo()` ? We're returning `true` anyway ?

Reply:
If something is wrong, getAlfrescoInfo() will throw an Exception (caught upper). Normally this contributor should always succeed. 
More specific contributors - to be implemented later - might fail, that's why the method returns a boolean. Example: failed license. 
Maybe it is not very standard, better suggestions are welcome.

Reply:
Instead of using exceptions, I would change the signature from `boolean isHealthy()` to `Health getHealth()`.

This `Health` class can have:
* a `String status` field (up/down) 
* a `Map<String, Object> details` 

this way you can put `error` -> `exception.getMessage()` in the details
+ we can expose the details later if the request is authenticated after all

Reply:
(authenticated after all -> with that I mean: when we manually check the /health endpoint, it could give more details, unauthenticated access still needs to be there)

Reply:
I pushed your requests.
But, I don't know how to have the same endpoint behaving differently depending on the authentication.